crates/esplora/src/async_ext.rs

```diff
@@ -85,10 +85,11 @@ impl EsploraAsyncExt for esplora_client::AsyncClient {
         local_tip: CheckPoint,
         request_heights: impl IntoIterator<IntoIter = impl Iterator<Item = u32> + Send> + Send,
     ) -> Result<local_chain::Update, Error> {
-        let request_heights = request_heights.into_iter().collect::<BTreeSet<_>>();
         let new_tip_height = self.get_height().await?;
 
-        // atomically fetch blocks from esplora
+        // Atomically fetch latest blocks from Esplora. This way, we avoid creating an update with
+        // an inconsistent set of blocks (assuming that a reorg depth cannot be greater than the
```

Comment:
What is this assumption I don't get it. There is no assumption being made here that I can see.

Reply:
If the reorg depth is greater than the latest block count that we fetched atomically, and we had to get another block lower than the atomic-fetch count... This additional block may not connect with the atomically-fetched blocks.